JetStreamDriver.js

```diff
@@ -570,16 +570,23 @@ class Benchmark {
     constructor(plan)
     {
         this.plan = plan;
-        this.testGroup = plan.testGroup;
-
+        this.tags = this._validateTags(plan.tags);
         this.iterations = getIterationCount(plan);
         this.isAsync = !!plan.isAsync;
-
         this.scripts = null;
-
         this._resourcesPromise = null;
     }
 
+    _validateTags(tags) {
+        const validated = new Set();
+        for (const tag of tags) {
+            if (!TAGS.has(tag))
+                throw new Error(`Invalid benchmark tag "${tag}"`);
```

Comment:
Nit: Seems a bit unnecessary since tags are mostly a debugging feature. We also only use the `TAGS` Set here so IMO I would skip this and remove `TAGS`. If you make that change you could just turn `this.tags = this._validateTags(plan.tags);` above into `this.tags = Set.from(plan.tags);` and optionally add 
```
if (this.tags.size != plan.tags.length) 
    throw new Error(`Duplicate tags ${plan.tags.filter((tag, index) => plan.tags.indexOf(tag) !== index)} in plan of ${this.name}`);
```

Reply:
I was mostly focused on preventing typos, but this can be just a unittest in the end (I'll re-add that after the cli test PR has landed).